renderer/src/routes/settings.route.tsx

```diff
@@ -0,0 +1,19 @@
+import { TitlePage } from '@/common/components/title-page'
+import { useSearch } from '@tanstack/react-router'
+import { SettingsTabs } from '@/common/components/settings/tabs/settings-tabs'
+import { Route } from './settings'
+
+export default function SettingsRouteComponent() {
+  const { tab } = useSearch({ from: Route.id })
+
+  if (typeof window === 'undefined' || !window.electronAPI) {
+    return null
```

Comment:
hmm, this seems weird... my understanding is that under some circumstances this would lead to an an empty page basically?

I understand that those circumstances should not really happen in real life, but it is still weird to show an empty page for an impossible scenario 🤔  why not throw an error for example?

at least I'm missing a code comment explaining what this logic is intended to solve because it's not obvious